tests/test_sparql.py

```diff
@@ -50,7 +50,32 @@ def test_sparql(self):
                             for result in results))
         self.assertTrue(all(result['citizen_name'].toPython() == karl.name
                             for result in results))
-        results = sparql(query, session=None)
+
+        def test_function(iri):
+            value = str(iri._value)
+            if value == "Freiburg":
+                return True
+            else:
+                return False
+
+        for row in results_core_session(citizen='cuds',
+                                        citizen_age=int,
+                                        citizen_name=str,
+                                        city_name=test_function):
+            self.assertTrue(row["citizen"].is_a(karl.oclass))
+            self.assertEqual(row["citizen_age"], karl.age)
+            self.assertEqual(row["citizen_name"], karl.name)
+            self.assertTrue(row["city_name"])
+
+        self.assertDictEqual(
```

Comment:
If I have learnt something dealing with the tests, is just test the external interface of a function or class, rather than the internal. Otherwise changing the internal behaviour afterwards is a pain. So I would just remove this and be happy when the above works.